include/api/CDataFrameAnalysisSpecification.h

```diff
@@ -40,9 +42,13 @@ class API_EXPORT CDataFrameAnalysisSpecification {
 public:
     using TStrVec = std::vector<std::string>;
     using TDataFrameUPtr = std::unique_ptr<core::CDataFrame>;
+    using TTemporaryDirectoryPtr = std::shared_ptr<core::CTemporaryDirectory>;
+    using TDataFrameUPtrTemporaryDirectoryPtrPr =
+        std::pair<TDataFrameUPtr, TTemporaryDirectoryPtr>;
     using TRunnerUPtr = std::unique_ptr<CDataFrameAnalysisRunner>;
     using TRunnerFactoryUPtr = std::unique_ptr<CDataFrameAnalysisRunnerFactory>;
     using TRunnerFactoryUPtrVec = std::vector<TRunnerFactoryUPtr>;
+    using TFatalErrorHandler = std::function<void(std::string)>;
```

Comment:
I think the argument here could be `const std::string&` rather than a value.

Reply:
The reason for this was that `std::ostringstream::str` (used in `HANDLE_FATAL`) returns by value so this gets copied when bound to the constant reference. I think the best solution is to move the string as below, which is an oversight, although this would never be a bottleneck.